internal/github.com/golang/gddo/httputil/negotiate.go

```diff
@@ -0,0 +1,36 @@
+// Copyright 2013 The Go Authors. All rights reserved.
+//
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file or at
+// https://developers.google.com/open-source/licenses/bsd.
+
+package httputil
+
+import (
+	"net/http"
+
+	"github.com/prometheus/client_golang/internal/github.com/golang/gddo/httputil/header"
+)
+
+// NegotiateContentEncoding returns the best offered content encoding for the
+// request's Accept-Encoding header. If two offers match with equal weight and
+// then the offer earlier in the list is preferred. If no offers are
+// acceptable, then "" is returned.
+func NegotiateContentEncoding(r *http.Request, offers []string) string {
```

Comment:
We have another problem.. Seems like this code somehow is exported to outside modules. Just tried using it in another module and managed to succeed... unfortunately. Somehow putting that under internal didn't help here 🤔 

![image](https://github.com/prometheus/client_golang/assets/6950331/d8458016-bac6-4d0a-b7f7-27b3090b352e)

![image](https://github.com/prometheus/client_golang/assets/6950331/e722feb6-961d-4afb-a8ab-d8e3ca7f8867)

![image](https://github.com/prometheus/client_golang/assets/6950331/fb6d0670-a87b-4c5f-930d-5e8355216c79)

We have to debug this a bit first 🙈

Reply:
I wonder... is this because we have some more directories under internal? We might want to read deeper how this work? Maybe it's because I use submodule (it should not matter in theory).

Reply:
Looks like it's expected as https://pkg.go.dev/cmd/go#hdr-Internal_Directories states. (: